feat: rephrase anomalous message

rahul-kriplani-jnpr opened the pull request:
refs: [COL-1730](https://paragon-automation.atlassian.net/browse/COL-1730)


AIOps marks  anomaly data point within the dynamic threshold after a cool down period so the messages seen to customers in this period were misleading so rephrased it to not have upper boundary, lower boundary and value in it. During aggregation as well these messages were misleading so having a generic message addresses that aswell.

And as we need to preserve the upper boundary and lower boundary in TSDB so have move the message in the normal term so that analytical engine still picks it up and saves it.

rahul-kriplani-jnpr committed:
feat: rephrase anomalous message

refs: COL-1730

rahul-kriplani-jnpr commented:
Please review. @gaddevijay and @sanjeevm-jsi

gaddevijay commented on juniper_official/chassis/check-fan-state-rpm.rule:
```diff
@@ -232,7 +232,7 @@ healthbot {
                         status {
                             color green;
-                            message "$fan-name RPM percent is $rpm-percent% is normal";
+                            message "$fan-name RPM percent($rpm-percent%) is between the predicted range of ($rpm-percent-anomaly-lower-boundary% and $rpm-percent-anomaly-upper-boundary%)";
```
Is the idea to move the ranges here, so that it stores in TSDB?

rahul-kriplani-jnpr replied:
Yes and since the messages are not displayed in normal state, these will not mislead.

sanjeevm-jsi commented:
Hi Rahul,
      When we are in waiting period, it is possible that the point marked normal I not within the boundaries. So, better to not call it as such unless we are differentiating.

Regards,
Sanjeev



Juniper Business Use Only

From: rahul-kriplani-jnpr ***@***.***>
Date: Friday, 14 November 2025 at 7:32 AM
To: Juniper/healthbot-rules ***@***.***>
Cc: Sanjeev Mishra ***@***.***>, Mention ***@***.***>
Subject: Re: [Juniper/healthbot-rules] feat: rephrase anomalous message (PR #1079)

[External Email. Be cautious of content]


@rahul-kriplani-jnpr commented on this pull request.

________________________________

In juniper_official/chassis/check-fan-state-rpm.rule<https://urldefense.com/v3/__https://github.com/Juniper/healthbot-rules/pull/1079*discussion_r2525508355__;Iw!!NEt6yMaO-gk!E_2E_5jrZUAQlpFwE2xDEj44IKsee_z-GKf5VR5X3HqtyqQxH_2xVRPXsj6UKku2N1KGeB16MjJXL8CSbppHeR_zng$>:

> @@ -232,7 +232,7 @@ healthbot {
                     then {
                         status {
                             color green;
-                            message "$fan-name RPM percent is $rpm-percent% is normal";
+                            message "$fan-name RPM percent($rpm-percent%) is between the predicted range of ($rpm-percent-anomaly-lower-boundary% and $rpm-percent-anomaly-upper-boundary%)";


Yes and since the messages are not displayed in normal state, these will not mislead.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/Juniper/healthbot-rules/pull/1079*discussion_r2525508355__;Iw!!NEt6yMaO-gk!E_2E_5jrZUAQlpFwE2xDEj44IKsee_z-GKf5VR5X3HqtyqQxH_2xVRPXsj6UKku2N1KGeB16MjJXL8CSbppHeR_zng$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/ATHHU3FFYD56TWKHVS5T5Z334UZZXAVCNFSM6AAAAACMASHAHOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTINRSGQ2TMOJWG4__;!!NEt6yMaO-gk!E_2E_5jrZUAQlpFwE2xDEj44IKsee_z-GKf5VR5X3HqtyqQxH_2xVRPXsj6UKku2N1KGeB16MjJXL8CSbppenfeEow$>.
You are receiving this because you were mentioned.

rahul-kriplani-jnpr commented:
> Hi Rahul,       When we are in waiting period, it is possible that the point marked normal I not within the boundaries. So, better to not call it as such unless we are differentiating. Regards, Sanjeev Juniper Business Use Only From: rahul-kriplani-jnpr ***@***.***> Date: Friday, 14 November 2025 at 7:32 AM To: Juniper/healthbot-rules ***@***.***> Cc: Sanjeev Mishra ***@***.***>, Mention ***@***.***> Subject: Re: [Juniper/healthbot-rules] feat: rephrase anomalous message (PR #1079) [External Email. Be cautious of content] @rahul-kriplani-jnpr commented on this pull request.
> ________________________________ In juniper_official/chassis/check-fan-state-rpm.rule<https://urldefense.com/v3/__https://github.com/Juniper/healthbot-rules/pull/1079*discussion_r2525508355__;Iw!!NEt6yMaO-gk!E_2E_5jrZUAQlpFwE2xDEj44IKsee_z-GKf5VR5X3HqtyqQxH_2xVRPXsj6UKku2N1KGeB16MjJXL8CSbppHeR_zng$>:
> @@ -232,7 +232,7 @@ healthbot {
> then { status { color green; - message "$fan-name RPM percent is $rpm-percent% is normal"; + message "$fan-name RPM percent($rpm-percent%) is between the predicted range of ($rpm-percent-anomaly-lower-boundary% and $rpm-percent-anomaly-upper-boundary%)"; Yes and since the messages are not displayed in normal state, these will not mislead. — Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/Juniper/healthbot-rules/pull/1079*discussion_r2525508355__;Iw!!NEt6yMaO-gk!E_2E_5jrZUAQlpFwE2xDEj44IKsee_z-GKf5VR5X3HqtyqQxH_2xVRPXsj6UKku2N1KGeB16MjJXL8CSbppHeR_zng$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/ATHHU3FFYD56TWKHVS5T5Z334UZZXAVCNFSM6AAAAACMASHAHOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTINRSGQ2TMOJWG4__;!!NEt6yMaO-gk!E_2E_5jrZUAQlpFwE2xDEj44IKsee_z-GKf5VR5X3HqtyqQxH_2xVRPXsj6UKku2N1KGeB16MjJXL8CSbppenfeEow$>. You are receiving this because you were mentioned.

@sanjeevm-jsi Right but in the waiting period as the anomaly status is not marked as 1, so this message will never be displayed on the UI.

rahul-kriplani-jnpr committed:
feat: refactor normal msg to avoid misleading during the wait period

refs: COL-1730

rahul-kriplani-jnpr committed:
feat: revert normal messages

Made changes from Insights side to always have upper and lower boundaries for the AIOps KPI, so we don't need to pass UB and LB in messages for them to be saved in TSDB.

refs: COL-1730

rahul-kriplani-jnpr commented:
> refs: [COL-1730](https://paragon-automation.atlassian.net/browse/COL-1730)
> 
> AIOps marks anomaly data point within the dynamic threshold after a cool down period so the messages seen to customers in this period were misleading so rephrased it to not have upper boundary, lower boundary and value in it. During aggregation as well these messages were misleading so having a generic message addresses that aswell.
> 
> And as we need to preserve the upper boundary and lower boundary in TSDB so have move the message in the normal term so that analytical engine still picks it up and saves it.

We now preserve UB and LB by default from the Insights side, so they no longer need to be included in the messages for storage in TSDB. Accordingly, the trigger messages have been cleaned up to exclude UB and LB, which eliminates the discrepancies previously observed in the UI.

rahul-kriplani-jnpr committed:
feat: removal current values from anomalous message

refs: COL-1730
